### PR TITLE
Edit string shortcut for model compilation

### PR DESCRIPTION
**Pull request by jaketae**
The document states that string identifiers can be used as shortcut, yet the example given is no different from the compilation method shown in the code block above (presumably without strings identifiers, which is also confusing given that the metric was defined with a string identifier). The edit adds clarity by demonstrating how models can be compiled with and without string identifiers.

### CODE REVIEW

File: site/en/guide/keras/train_and_evaluate.ipynb [hunk 1]
```diff
@@ -375,8 +375,8 @@
       },
       "outputs": [],
       "source": [
-        "model.compile(optimizer=keras.optimizers.RMSprop(learning_rate=1e-3),\n",
-        "              loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),\n",
```
Comment (lamberta) [inline site/en/guide/keras/train_and_evaluate.ipynb]: @MarkDaoust I think `from_logits=True` is the preferred usage these days, yes?

Comment (MarkDaoust) [reply site/en/guide/keras/train_and_evaluate.ipynb]: Exactly.

I reverted that line, and updated the text.
